packages/manager/apps/web/client/app/domain/zone/zone.routing.js

```diff
@@ -1,13 +1,137 @@
-  const commonResolve = {
-    goToZoneHistory: /* @ngInject */ ($state) => (params) =>
-      $state.go('app.zone.details.zone-history', params),
-    breadcrumb: /* @ngInject */ ($translate) =>
-      $translate.instant('domain_zone'),
-  };
+  $stateProvider.state('app.domain.product.zone-history', {
+    url: '/zone-history',
+    views: {
+      domainView: {
+        component: 'domainZoneDashboardHistory',
+      },
+    },
+    resolve: {
+      ...commonResolveForZoneHistory,
+      goToDnsData: /* @ngInject */ ($state) => (url) =>
+        $state.go('app.domain.product.zone-history.view', { url }),
+      goToDnsRestore: /* @ngInject */ ($state) => (url, chosenDate) =>
+        $state.go('app.domain.product.zone-history.restore', {
+          url,
+          chosenDate,
+        }),
+      goBack: /* @ngInject */ ($state, $stateParams) => () =>
+        $state.go('app.domain.product.zone', $stateParams),
+      goToDiffViewer: /* @ngInject */ ($state) => (
+        dnsEntriesForComparison,
+        zoneName,
+      ) =>
+        $state.go('app.domain.product.diff-tool-viewer', {
+          selectedDates: dnsEntriesForComparison
+            .filter((u) => u.active)
+            .map((u) => u.date),
+          productId: zoneName,
+        }),
+    },
+    params: {
+      productId: null,
+    },
+  });
+
+  $stateProvider.state('app.domain.product.diff-tool-viewer', {
+    url: '/diff-tool-viewer',
+    views: {
+      domainView: {
+        component: 'domainZoneDiffToolViewerHistory',
+      },
+    },
+    resolve: {
+      ...commonResolveForZoneHistory,
+      goBack: /* @ngInject */ ($state, $stateParams) => () =>
+        $state.go('app.domain.product.zone-history', $stateParams),
+    },
+    params: {
+      selectedDates: null,
+      productId: null,
+    },
+  });
+
+  $stateProvider.state('app.domain.product.zone-history.view', {
+    url: '/view',
+    layout: { name: 'modal', keyboard: true },
+    views: {
+      modal: {
+        component: 'zoneHistoryView',
+      },
+    },
+    params: {
+      url: {
+        type: 'string',
+        value: null,
+      },
+    },
+    resolve: {
+      ...commonResolveForZoneHistory,
+      url: /* @ngInject */ ($stateParams) => $stateParams.url,
+      goBack: /* @ngInject */ ($state) => () => $state.go('^'),
+      getDnsZoneData: /* @ngInject */ (DNSZoneService) => (url) =>
+        DNSZoneService.getDnsFile(url),
+    },
+  });
+
+  $stateProvider.state('app.domain.product.zone-history.restore', {
+    url: '/restore',
+    layout: { name: 'modal', keyboard: true },
+    views: {
+      modal: {
+        component: 'zoneHistoryRestore',
+      },
+    },
+    params: {
+      chosenDate: {
+        type: 'string',
+        value: null,
+      },
+    },
+    resolve: {
+      ...commonResolveForZoneHistory,
+      chosenDate: /* @ngInject */ ($stateParams) => $stateParams.chosenDate,
+      zoneId: /* @ngInject */ ($stateParams) => $stateParams.productId,
```

Comment:
**nitpick:** Use `$transition` instead.
```suggestion
      chosenDate: /* @ngInject */ ($transition$) => $transition$.params().chosenDate,
      zoneId: /* @ngInject */ ($transition$) => $transition$.params().productId,
```

Reply:
Allright I can change here but can I understand why transition is prefered over stateparams ?

Reply:
Ah OK found it here: https://github.com/angular-ui/ui-router/issues/3237

StateParams seems deprecated.